WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -29,6 +29,9 @@ class CompetitionsController < ApplicationController
 
   def new
     @competition = Competition.new
+    if current_user.any_kind_of_delegate?
+      @competition.delegates |= [current_user]
```

Comment:
I tried looking into this stuff, and it's fairly contentious and confusing. It seems Rails does this right with direct associations, if you specify :uniq on the relation, but not on `through` relations, and that's (debatably) a bug.

Or something... This feels like a bottomless hole so I stopped digging.

https://github.com/rails/rails/issues/8573